src/description/description/common.xacro

```diff
@@ -0,0 +1,232 @@
+        <color rgba="0.2 0.2 1 1"/>
+    </material>
+
+    <material name="black">
+        <color rgba="0 0 0 1"/>
+    </material>
+
+    <material name="yellow">
+        <color rgba="1.0 1.0 0.0 1.0"/>
+    </material>
+
+    <material name="green">
+        <color rgba="0.0 1.0 0.0 1.0"/>
+    </material>
+
+  <!-- Properties -->
+    <!-- Stepper Motor Properties -->
+        <xacro:property name="stepper_min_angle" value="0"/>
+        <xacro:property name="stepper_max_angle" value="6.28"/>
+
+    <!-- Carriage Properties -->
+        <xacro:property name="carriage_min_position" value="-0.68"/>
+        <xacro:property name="carriage_max_position" value="0.68"/>
+        <xacro:property name="carriage_stepper_offset" value="${carriage_max_position}"/> <!-- Assuming a min angle of 0 rad for the stepper -->
+        <xacro:property name="carriage_stepper_multiplier" value="${(carriage_min_position - carriage_stepper_offset) / stepper_max_angle}"/>
+        
+    <!-- Hand Properties -->
+        <xacro:property name="hand_length" value="0.2"/>
+        <xacro:property name="hand_width" value="0.1"/>
+        <xacro:property name="hand_height" value="0.05"/>
+
+    <!-- Finger Properties -->
+        <xacro:property name="finger_length" value="0.1"/>
+        <xacro:property name="finger_thickness" value="0.01"/>
+        <xacro:property name="finger_spacing" value="${hand_length/6}"/>
+        <xacro:property name="thumb_offset" value="${-hand_length/2 + finger_spacing}"/>
+        <xacro:property name="index_offset" value="${-hand_length/2 + 2*finger_spacing}"/>
+        <xacro:property name="middle_offset" value="0"/>
+        <xacro:property name="ring_offset" value="${hand_length/2 - 2*finger_spacing}"/>
+        <xacro:property name="pinky_offset" value="${hand_length/2 - finger_spacing}"/>
+
+  <!-- Macros -->
+    <!-- Stepper Motor Macro -->
+        <xacro:macro name="stepper" params="name parent_link origin_xyz">
+            <link name="${name}">
+                <visual>
+                    <geometry>
+                        <cylinder radius="0.02" length="0.05"/>
+                    </geometry>
+                    <material name="black"/>
+                </visual>
+                <inertial>
+                    <mass value="0.1"/>
+                    <origin xyz="0 0 0"/>
+                    <inertia ixx="0.0001" ixy="0.0" ixz="0.0" iyy="0.0001" iyz="0.0" izz="0.0001"/>
+                </inertial>
+                <gazebo>
+                    <material>Gazebo/Black</material>
+                </gazebo>
+            </link>
+            <joint name="${name}_joint" type="revolute">
+                <parent link="${parent_link}"/>
+                <child link="${name}"/>
+                <origin xyz="${origin_xyz}" rpy="0 0 0"/>
+                <axis xyz="0 0 1"/>
+                <limit lower="${stepper_min_angle}" upper="${stepper_max_angle}" effort="5" velocity="1.0"/>
+            </joint>
+        </xacro:macro>
```

Comment:
How would the steppers provides actuation to the prismatic joints if the steppers are fixed? Also that will make the joints used by the JTCs very confusing because it will not be controlling the stepper anymore but instead the carriages.

I thought of what you said before I had to stepup things and complicate things a little but make it more intuitive and closer to what the reality will be.